internal/keystore/azure/key-vault.go

```diff
@@ -107,24 +113,19 @@ func (s *Store) Create(ctx context.Context, name string, value []byte) error {
 	if err != nil {
 		return fmt.Errorf("azure: failed to create '%s': %v", name, err)
 	}
-	if stat.StatusCode == http.StatusConflict && stat.ErrorCode == "ObjectIsDeletedButRecoverable" {
-		stat, err = s.client.PurgeSecret(ctx, name)
+	if stat.StatusCode == http.StatusConflict && (stat.ErrorCode == "ObjectIsDeletedButRecoverable" || stat.ErrorCode == "ObjectIsBeingDeleted") {
+		stat, err = s.purgeWithRetry(ctx, name, 25)
 		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
 			return err
 		}
 		if err != nil {
 			return fmt.Errorf("azure: failed to create '%s': failed to purge deleted secret: %v", name, err)
 		}
-		if stat.StatusCode != http.StatusNoContent {
+		if stat.StatusCode != http.StatusOK {
 			return fmt.Errorf("azure: failed to create '%s': failed to purge deleted secret: %s (%s)", name, stat.Message, stat.ErrorCode)
 		}
 
-		const (
-			Retry  = 7
-			Delay  = 200 * time.Millisecond
-			Jitter = 800 * time.Millisecond
-		)
-		for i := 0; i < Retry; i++ {
+		for i := 0; i < 7; i++ {
```

Comment:
If we we use `s.purgeWithRetry(ctx, name, 25)`, it's better to remove the retry.

Reply:
Even after purging, there is sometimes a bit of (unpredictable) latency before you can actually create a new secret. That's why I kept the retry mechanism here.